packages/shorebird_cli/lib/src/shorebird_env.dart

```diff
@@ -51,7 +51,18 @@ class ShorebirdEnv {
   ///
   /// Assumes we are running from $ROOT/bin/cache.
   Directory get shorebirdRoot {
-    return File(platform.script.toFilePath()).parent.parent.parent;
+    final script = platform.script.toFilePath();
+    final lastPathComponent = p.basename(p.dirname(script));
+    // We're being run directly (e.g. `dart run shorebird_cli`)
+    if (lastPathComponent == 'bin' || lastPathComponent == 'tool') {
+      return File(script).parent.parent.parent.parent;
+    }
+    if (lastPathComponent != 'cache') {
+      throw StateError('Unexpected script path: $script');
+    }
```

Comment:
Not sure it's worth supporting multiple entrypoints.
If you want to run a local version you can `dart pub global activate --source path ./path/to/shorebird_cli` rather than changing the assumptions to also support `dart run shorebird_cli/bin/...`